packages/storage/amplify_storage_s3_dart/lib/src/amplify_storage_s3_dart_impl.dart

```diff
@@ -149,16 +163,30 @@ class AmplifyStorageS3Dart extends StoragePluginInterface<
     }
 
     dependencyManager
+      ..addBuilder<AppPathProvider>(S3DartAppPathProvider.new)
+      ..addBuilder(
+        transfer.TransferDatabase.new,
+        const Token<transfer.TransferDatabase>([Token<AppPathProvider>()]),
+      )
       ..addInstance<StorageS3Service>(
         StorageS3Service(
           credentialsProvider: credentialsProvider,
           defaultBucket: s3pluginConfig.bucket,
           defaultRegion: s3pluginConfig.region,
           prefixResolver: _prefixResolver!,
           logger: logger,
+          dependencyManager: dependencyManager,
         ),
-      )
-      ..addInstance<AppPathProvider>(const S3DartAppPathProvider());
+      );
+
+    // Run this later
+    scheduleMicrotask(() async {
+      await Amplify.asyncConfig;
+      if (_zIsTest) {
```

Comment:
Can you add a TODO to remove this? storageS3Service should be injected and this logic shouldn't depend on whether we're in a test